java-lib/src/main/java/org/clink/feature/onehotencoder/ClinkOneHotEncoderModel.java

```diff
@@ -0,0 +1,136 @@
+                new RowTypeInfo(
+                        ArrayUtils.addAll(
+                                inputTypeInfo.getFieldTypes(),
+                                Collections.nCopies(
+                                                outputCols.length,
+                                                ExternalTypeInfo.of(Vector.class))
+                                        .toArray(new TypeInformation[0])),
+                        ArrayUtils.addAll(inputTypeInfo.getFieldNames(), outputCols));
+
+        StreamTableEnvironment tEnv =
+                (StreamTableEnvironment) ((TableImpl) inputs[0]).getTableEnvironment();
+
+        DataStream<Row> output =
+                tEnv.toDataStream(inputs[0])
+                        .map(
+                                new GenerateOutputsFunction(modelPath, getInputCols()),
+                                outputTypeInfo);
+
+        return new Table[] {tEnv.fromDataStream(output)};
+    }
+
+    private static class GenerateOutputsFunction extends RichMapFunction<Row, Row> {
+        private final String modelPath;
+        private Pointer modelPointer;
+        private final String[] inputCols;
+
+        private GenerateOutputsFunction(String modelPath, String[] inputCols) {
+            this.modelPath = modelPath;
+            this.inputCols = inputCols;
+        }
+
+        @Override
+        public void open(Configuration parameters) throws Exception {
+            super.open(parameters);
+            this.modelPointer = ClinkKernels.INSTANCE.OneHotEncoderModel_load(modelPath);
+        }
+
+        @Override
+        public Row map(Row row) {
+            Row resultRow = new Row(inputCols.length);
+            for (int i = 0; i < inputCols.length; i++) {
+                String inputCol = inputCols[i];
+                int number = ((Number) row.getField(inputCol)).intValue();
+                ClinkSparseVector.ByValue vectorCpp =
+                        ClinkKernels.INSTANCE.OneHotEncoderModel_transform(modelPointer, number, i);
+                SparseVector vector = vectorCpp.toSparseVector();
+                resultRow.setField(i, vector);
+            }
+            return Row.join(row, resultRow);
+        }
+
+        @Override
+        public void close() throws Exception {
+            super.close();
+            ClinkKernels.INSTANCE.OneHotEncoderModel_delete(modelPointer);
+        }
+    }
+
+    @Override
+    public void save(String path) throws IOException {
+        // Clink Operators need not support saving for now.
+        throw new UnsupportedEncodingException();
+    }
+
+    public static ClinkOneHotEncoderModel load(StreamExecutionEnvironment env, String path)
+            throws IOException {
+        OneHotEncoderModel model = ReadWriteUtils.loadStageParam(path);
+        ClinkOneHotEncoderModel clinkModel = new ClinkOneHotEncoderModel();
+        ReadWriteUtils.updateExistingParams(clinkModel, model.getParamMap());
+        clinkModel.modelPath = path;
```

Comment:
Normally users expect the IO operation (e.g. reading model data from the given path) to happen during `load(...)` instead of `transform(...)`. Otherwise the behavior of the program might be surprising to user which makes the program harder to debug.

How about we do `OneHotEncoderModel_load(...)` inside the `load(...)` here and pass the resulting pointer to the model via e.g. `OneHotEncoderModel::setUnderlyingModel(...)`?

Reply:
As discussed offline, loading C++ operator in `load()`, which is at client rather than server side, is not proper. Besides, we also need to call `save()` in `transform()` if `ClinkOneHotEncoderModel.transform()` is called right after the estimator's `fit()` method is invoked. Thus so far by now we still need to to the loading process in `transform()`.